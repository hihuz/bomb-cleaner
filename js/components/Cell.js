import React from 'react';

const Cell = ({ handleLeftClick, handleRightClick, index, opened, flagged, value }) => {
  const colors = ['#000', '#0100fe', '#017f01', '#fe0000', '#010080', '#810102', '#008081', '#000', '#808080'];
  const styles = {
    color: value !== 'X' && value !== ' ' && opened ? colors[value] : colors[0]
  };
  function leftClick(e) {
    e.preventDefault();
    if (handleLeftClick) { handleLeftClick(index, value); }
  }
  function rightClick(e) {
    e.preventDefault();
    if (handleRightClick) { handleRightClick(index); }
  }
  return (
    <div
      className={`cell${opened ? ' opened' : ''}${value === 'X' && opened ? ' bomb' : ''}`}
      onClick={leftClick}
      onContextMenu={rightClick}
      style={styles}
    >
      {opened && value !== 'X' ? value : ' '}
      {opened && value === 'X' ? <i className="icon-bomb" /> : ' '}
      {flagged && !opened ? <i className="icon-flag" /> : ''}
    </div>
  );
};

export default Cell;
