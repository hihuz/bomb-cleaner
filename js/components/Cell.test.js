import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Cell from './Cell';

const evt = {
  preventDefault() {
  }
};

test('Snapshot', () => {
  const component = shallow(<Cell />);
  const tree = shallowToJson(component);
  expect(tree).toMatchSnapshot();
});

test('Should display as an open cell & value displayed if opened props', () => {
  const component = shallow(<Cell opened={true} value={' '} />);
  expect(component.hasClass('opened')).toEqual(true);
  expect(component.text()).toEqual(' ');
  expect(component.find('i').length).toEqual(0);
});


test('Should display as an open bomb if opened props & "X" value props', () => {
  const component = shallow(<Cell opened={true} value={'X'} />);
  expect(component.hasClass('opened')).toEqual(true);
  expect(component.hasClass('bomb')).toEqual(true);
  expect(component.find('i').length).toEqual(1);
  expect(component.find('i').hasClass('icon-bomb')).toEqual(true);
  expect(component.find('i').hasClass('icon-flagged')).toEqual(false);
});

test('Should display as a flagged cell if flagged props', () => {
  const component = shallow(<Cell opened={false} value={'X'} flagged={true} />);
  expect(component.text()).toEqual(' ');
  expect(component.hasClass('opened')).toEqual(false);
  expect(component.find('i').length).toEqual(1);
  expect(component.find('i').hasClass('icon-bomb')).toEqual(false);
  expect(component.find('i').hasClass('icon-flag')).toEqual(true);
});

test('Should NOT display as a flagged cell if flagged props and opened props', () => {
  const component = shallow(<Cell opened={true} value={3} flagged={true} />);
  expect(component.text()).toEqual('3');
  expect(component.hasClass('opened')).toEqual(true);
  expect(component.find('i').length).toEqual(0);
});

test('Should call the function passed as handleLeftClick props if the cell is left-clicked', () => {
  const fakeClickHandler = jest.fn();
  const component = shallow(<Cell handleLeftClick={fakeClickHandler} />);
  component.simulate('click', evt);
  expect(fakeClickHandler).toHaveBeenCalledTimes(1);
});

test('Should call the function passed as handleRightClick props if the cell is right-clicked', () => {
  const fakeClickHandler = jest.fn();
  const component = shallow(<Cell handleRightClick={fakeClickHandler} />);
  component.simulate('contextmenu', evt);
  expect(fakeClickHandler).toHaveBeenCalledTimes(1);
});
