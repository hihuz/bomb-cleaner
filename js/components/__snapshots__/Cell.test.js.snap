exports[`test Snapshot 1`] = `
<div
  className="cell"
  onClick={[Function]}
  onContextMenu={[Function]}
  style={
    Object {
      "color": "#000",
    }
  }>
   
</div>
`;
